crates/astria-sequencer-relayer/src/data_availability.rs

```diff
@@ -168,6 +168,33 @@ pub struct RollupNamespaceData {
 
 impl NamespaceData for RollupNamespaceData {}
 
+impl RollupNamespaceData {
+    pub fn new(
+        block_hash: Hash,
+        chain_id: Vec<u8>,
+        rollup_txs: Vec<Vec<u8>>,
+        inclusion_proof: InclusionProof,
+    ) -> Self {
+        Self {
+            block_hash,
+            chain_id,
+            rollup_txs,
+            inclusion_proof,
+        }
+    }
+
+    pub fn verify_inclusion_proof(&self, root_hash: [u8; 32]) -> eyre::Result<()> {
+        let rollup_data_tree = MerkleTree::from_leaves(self.rollup_txs.clone());
+        let rollup_data_root = rollup_data_tree.root();
+        let mut leaf = self.chain_id.clone();
+        leaf.append(&mut rollup_data_root.to_vec());
+
+        self.inclusion_proof
+            .verify(&leaf, root_hash)
+            .wrap_err("failed to verify rollup data inclusion proof")
```

Comment:
Nit: you can just return the result of `self.inclusion_proof.verify(&leaf, root_hash)`, no need to wrap this in another error. When used you'd otherwise end up with something like this:
```rust
my_rollup_data.verify_inclusion_proof(some_hash)
    .wrap_err("failed to verify inclusion proof")?;
```
And then the error would be e.g.:
```
Error occured:
0: failed to verify inclusion proof

1. failed to verify rollup data inclusion proof

2. <actual errors>
```